printf/stdio_printf.c

```diff
@@ -0,0 +1,2645 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - stdio.h
+ *
+ * TESTED:
+ *    - dprintf()
+ *    - fprintf()
+ *    - printf()
+ *    - snprintf()
+ *    - sprintf()
+ *    - vdprintf()
+ *    - vfprintf()
+ *    - vprintf()
+ *    - vsnprintf()
+ *    - vsprintf()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <float.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <unity_fixture.h>
+#include <signal.h>
+
+static FILE *output_file;
+static char buff[700];
+
+enum OutputType { DTEST,
+	FTEST,
+	TEST,
+	SNTEST,
+	STEST };
+
+#define NUM_OUTPUT_TYPES 5
+
+#define PATH "stdio_printf_test"
+
+/* comma delimiter macro ',' for multiple arguments parsing to *printf() function via macros */
+#define COMMA ,
+
+/* macro assertions*/
+
+#define printf_test(format, value, expect) \
```

Comment:
Let's have a common convention in our test and prefix all global functions/variables by `test` and use camelCase as it's recommended in our coding convention.

```suggestion
static FILE *test_outFile;
static char test_buff[700];

enum test_outType { DTEST,
	FTEST,
	TEST,
	SNTEST,
	STEST };

#define NUM_OUTPUT_TYPES 5

#define PATH "stdio_printf_test"

/* comma delimiter macro ',' for multiple arguments parsing to *printf() function via macros */
#define COMMA ,

/* macro assertions*/

#define test_assertPrintfs(format, value, expect) \
```